impersonate/impersonate_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package impersonate
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/api/option"
+)
+
+func TestTokenSource_serviceAccount(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		name            string
+		targetPrincipal string
+		scopes          []string
+		lifetime        time.Duration
+		wantErr         bool
+	}{
+		{
+			name:    "missing targetPrincipal",
+			wantErr: true,
+		},
+		{
+			name:            "missing scopes",
+			targetPrincipal: "foo@project-id.iam.gserviceaccount.com",
+			wantErr:         true,
+		},
+		{
+			name:            "lifetime over max",
+			targetPrincipal: "foo@project-id.iam.gserviceaccount.com",
+			scopes:          []string{"scope"},
+			lifetime:        3601 * time.Second,
+			wantErr:         true,
+		},
+		{
+			name:            "lifetime over max",
+			targetPrincipal: "foo@project-id.iam.gserviceaccount.com",
+			scopes:          []string{"scope"},
+			wantErr:         false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Same `tt.name` comment. I don't think it's strictly necessary since these aren't run in parallel, though.

Reply:
Ack.